tests/unit/test_stats.py

```diff
@@ -36,21 +36,26 @@ def test_rolling_window_find_quantiles():
     desired_quantiles = [0.4, 0.5, 0.6]
     # We have points at 0 and 1 in both x and y. We set the decay length to 20 so 0
     # and 1 get weightings of 1/3 at their own end and 1/6 at the other end.
-    # This then means gradient of either 3 or 6, depending on whether you encounter the
-    # 3 first (nearer 0, which occurs first above) or 6 first (nearer 1, occurs later)
+    # Subtracting half from each of these, for quantiles above 5/12 we get q - 5/12
```

Comment:
I'm not sure I fully follow: 5/12 = 1/2 - 1/2*1/6 and 4/6 = 2/3 = 1 - 1/3 ? Where does the 'subtracting half from each' come in?

Reply:
Because we associate the value y with a cumulative sum of up to half the weight. It's the cumsum(w(x)) - w(x) /2.